src/Features/CSharp/Portable/SignatureHelp/AbstractCSharpSignatureHelpProvider.cs

```diff
@@ -68,5 +96,302 @@ protected static SymbolDisplayPart NewLine()
+                        var namedParameterIndex = parameters.IndexOf(p => p.Name == name);
+                        if (namedParameterIndex < 0)
+                        {
+                            return false;
+                        }
+
+                        if (namedParameterIndex != currentParameterIndex)
+                        {
+                            seenOutOfPositionArgument = true;
+                        }
+
+                        AddArgToParamMapping(argumentIndex, namedParameterIndex);
+                        IncrementParameterIndexIfNeeded();
+                    }
+                    else if (IsEmptyArgument(argument.Expression))
+                    {
+                        if (!seenOutOfPositionArgument)
+                        {
+                            // We count the empty argument as a used position
+                            AddArgToParamMapping(argumentIndex, currentParameterIndex);
+                            IncrementParameterIndexIfNeeded();
+                        }
+                    }
+                    else if (seenOutOfPositionArgument)
+                    {
+                        // Unnamed arguments are not allowed after an out-of-position argument
+                        return false;
+                    }
+                    else
+                    {
+                        AddArgToParamMapping(argumentIndex, currentParameterIndex);
+                        IncrementParameterIndexIfNeeded();
+                    }
+                }
+
+                return true;
+
+                void IncrementParameterIndexIfNeeded()
+                {
+                    if (!seenOutOfPositionArgument && !inParams)
+                    {
+                        currentParameterIndex++;
+                    }
+                }
+
+                void AddArgToParamMapping(int argumentIndex, int parameterIndex)
+                {
+                    Debug.Assert(parameterIndex >= 0);
+                    Debug.Assert(parameterIndex < parameterCount);
+
+                    if (parameters[parameterIndex].IsParams)
+                    {
+                        inParams = true;
+                    }
+
+                    argToParamMap[argumentIndex] = parameterIndex;
+                }
+            }
+
+            private static bool IsEmptyArgument(ExpressionSyntax expression)
+                => expression is IdentifierNameSyntax { Identifier.ValueText.Length: 0 };
+
+            /// <summary>
+            /// Given the cursor position, find which argument is active.
+            /// This will be useful to later find which parameter should be highlighted.
+            /// </summary>
+            internal static int TryGetArgumentIndex(SeparatedSyntaxList<ArgumentSyntax> arguments, int position)
+            {
+                if (arguments.Count == 0)
+                {
+                    return -1;
+                }
+
+                if (position < arguments.Span.Start)
```

Comment:
I'll see if this could be removed, or handled as an error rather than like the case above.